changes/2024-05-20-keystore-kms-config/change.md

```diff
@@ -0,0 +1,274 @@
+such as the AWS Encryption SDK or AWS Database Encryption SDK.
+
+A "discovery Keystore" is a (Branch) Keystore with
+a KMS Configuration of Discovery or Multi-Region Discovery (MRDiscovery).
+
+A "static Keystore" is a (Branch) Keystore with
+a KMS Configuration of KMS Key ARN.
+
+A "strict Keystore" is a (Branch) Keystore with
+a KMS Configuration of KMS Multi-Region Key ARN or KMS Key ARN.
+
+### Conventions used in this document
+
+The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+Allow MPL Consumers to easily configure
+a Hierarchical Keyring to use
+the KMS Key ARNs already persisted to a Logical Keystore.
+
+Additionally, ensure a KMS Alias is not permitted as a KMS Key ARN.
+
+## Out of Scope
+
+- Administrative/Mutating operations for a Keystore with a KMS Configuration of Discovery;
+  these are `VersionKey` & `CreateKey`.
+- Lifecycle operations for the Keystore
+- Changes to the Hierarchy Keyring to improve Caching performance across multiple Keystores
+
+## Motivation
+
+The (Branch) Keystore and the AWS KMS Hierarchical Keyring
+can be used to facilitate Client Side Encryption for
+a Multi-tenant environment where each tenant has a unique
+[Branch Key](../../framework/structures.md#branch-key-materials).
+However, Tenants may want an MPL Consumer to protect
+their data with a particular KMS Key that is NOT necessarily the same
+as other tenants.
+
+Such a Multi-KMS Keystore is, prior to this change, difficult to use
+as a Keystore instance is restricted to one and only one KMS Key ARN.
+Many MPL Consumers have developed a work around to this restriction
+by constructing and then caching a Keystore per KMS Key ARN.
+These MPL Consumers than stand up their own "request-to-Keystore-to-BranchKeyID" logic.
+This workaround may require an additional lookup
+compared to the normal "Encryption Context determines Branch Key ID"
+path.
+
+The goal then is to allow MPL Consumers to construct Keystore instances that
+MAY use the KMS Key ARNs already persisted to a Logical Keystore.
+
+Put another way,
+usage of Branch Keys need only consider the Keystore's Logical Name
+(and therefore, backing persistance medium, which is currently always a DynamoDB table)
+and Branch Key ID.
+
+Usage may not need the KMS ARN.
+
+### KMS Alias
+
+Because the Branch Key creation operation asserted the result
+of KMS `ReEncrypt` matched the Keystore's configured KMS Key ARN,
+a KMS Key Alias could never be used with a Keystore to successfully create a Branch Key.
+
+However, the runtime exception thrown did not make it clear
+why `CreateKey` failed.
+
+Rather than correct the exception,
+Keystore construction MUST fail if the KMS Configuration
+is "strict"
+(KMS Key ARN or KMS MRKey ARN)
+and the provided ARN is an Alias.
```

Comment:
Why not make the Alias work? We should add some words about why we chose the error path

Reply:
What wording is missing from lines 102 to 107 that you would like to add?